docs/vectorizer/worker.md

```diff
@@ -87,94 +170,114 @@ On your local machine:
 
-1. **Run the vectorizer worker**
+If you are using an embedding provider that requires an API key (which most do),
+you can set the API key through an environment variable or a .env file. We recommend
+using a .env file to set the API key.
 
-    After you [define a vectorizer in your database](/docs/vectorizer.md#define-a-vectorizer), you run 
-    a vectorizer worker to generate and update your embeddings:
+For example, if you are using OpenAI, you can set the API key in a .env file as follows:
+
+```
+OPENAI_API_KEY=<your-openai-api-key>
+```
+
+Or you can set the API key through an environment variable:
+
+```
+export OPENAI_API_KEY=<your-openai-api-key>
+```
 
-    1. Configure environment variables if necessary (see [Additional configuration via environment variables](#additional-configuration-via-environment-variables))
-       for a list of the available environment variables.
 
-    1. Run the vectorizer worker:
+## Advanced configuration options
 
-       ```shell
-       pgai vectorizer worker -d <db-connection-string>
-       ```
+Most users of the vectorizer worker will be happy with the default configuration for all other options.
+But, if you need to, you can control the following:
 
-## Run vectorizers with vectorizer worker
+- The vectorizer ids that are processed by the vectorizer worker ([section below](#control-which-vectorizers-are-processed))
+- The time between vectorizer worker runs ([section below](#set-the-time-between-vectorizer-worker-runs))
+- The number of asynchronous tasks running in a vectorizer worker ([section below](#set-the-number-of-asynchronous-tasks-running-in-a-vectorizer-worker))
+- Whether to run the vectorizer worker once and then exit ([section below](#run-the-vectorizer-worker-once-and-then-exit))
+
+All of these options can be set through the command line arguments, environment variables, or through an argument to the `Worker` 
+class constructor in the `pgai` Python package.
+
+| Option | Command line argument | Environment variable | `Worker` class constructor argument |
+|--------|-----------------------|--------------------|-------------------------------------|
+| Control which vectorizers are processed | `-i` / `--vectorizer-id` | `PGAI_VECTORIZER_WORKER_VECTORIZER_IDS` | `vectorizer_ids` |
+| Set the time between vectorizer worker runs | `--poll-interval` | `PGAI_VECTORIZER_WORKER_POLL_INTERVAL` | `poll_interval` |
+| Set the number of asynchronous tasks running in a vectorizer worker | `-c` / `--concurrency` | `PGAI_VECTORIZER_WORKER_CONCURRENCY` | `concurrency` |
+| Run the vectorizer worker once and then exit | `--once` | `PGAI_VECTORIZER_WORKER_ONCE` | `once` |
+
+
+### Control which vectorizers are processed
+
+If you want to run a vectorizer worker that only processes a subset of the vectorizers in your database,
+you can do so by specifying the vectorizer ids you want to process. You can do this by using the 
+`-i` / `--vectorizer-id` command line argument. 
```

Comment:
This should actually also take vectorizer names.

Reply:
Yup but leaving for later